RecoLocalCalo/CaloTowersCreator/src/CaloTowersCreationAlgo.cc

```diff
@@ -472,72 +491,75 @@ void CaloTowersCreationAlgo::assignHitHcal(const CaloRecHit * recHit) {
   // SPECIAL handling of tower 28 merged depths --> half into tower 28 and half into tower 29
   if (detId.det()==DetId::Hcal && 
       HcalDetId(detId).subdet()==HcalEndcap &&
-	  (theHcalPhase==0 || theHcalPhase==1) &&
-	  std::find(mergedDepths.begin(), mergedDepths.end(), HcalDetId(detId).depth())!=mergedDepths.end() &&
-      //HcalDetId(detId).depth()==3 &&
+      (theHcalPhase==0 || theHcalPhase==1) &&
       HcalDetId(detId).ietaAbs()==theHcalTopology->lastHERing()-1) {
+    bool merge = (subdetOne == HcalEndcap && (std::find(phizOne.begin(),phizOne.end(),std::pair<int,int>(HcalDetId(detId).iphi(),HcalDetId(detId).zside())) == phizOne.end())) ? 
+      (std::find(mergedDepthsOne.begin(), mergedDepthsOne.end(), HcalDetId(detId).depth())!=mergedDepthsOne.end()) :
+      (std::find(mergedDepths.begin(), mergedDepths.end(), HcalDetId(detId).depth())!=mergedDepths.end());
```

Comment:
please break it up into several booleans or with if-else for better readability.